server/src/main/java/org/elasticsearch/cluster/coordination/LeaderChecker.java

```diff
@@ -234,15 +234,21 @@ public void handleException(TransportException exp) {
                         }
 
                         if (exp instanceof ConnectTransportException || exp.getCause() instanceof ConnectTransportException) {
-                            logger.debug(new ParameterizedMessage("leader [{}] disconnected, failing immediately", leader), exp);
+                            if (logger.isDebugEnabled()) {
+                                logger.debug(new ParameterizedMessage(
+                                    "leader [{}] disconnected during check, restarting discovery", leader), exp);
+                            } else {
+                                logger.info("leader [{}] disconnected during check, restarting discovery", leader);
```

Comment:
Instead of logging here at info level, I would prefer to pass the necessary info (message + exception) to `Coordinator.getOnLeaderFailure` and let that one do the logging if it effectively turns the node back to candidate.
Also I would prefer not to use the "leader" terminology when logging at info level.

Reply:
Ok, I've moved the logging to the `Coordinator` in bd0c29cedd0465e4d0baa55c3dcec46a935b501c.